contracts/registries/GroupIPAssetRegistry.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.23;
+
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import { IIPAccount } from "../interfaces/IIPAccount.sol";
+import { IGroupIPAssetRegistry } from "../interfaces/registries/IGroupIPAssetRegistry.sol";
+import { IGroupingModule } from "../interfaces/modules/grouping/IGroupingModule.sol";
+import { ProtocolPausableUpgradeable } from "../pause/ProtocolPausableUpgradeable.sol";
+import { Errors } from "../lib/Errors.sol";
+import { IPAccountStorageOps } from "../lib/IPAccountStorageOps.sol";
+
+/// @title GroupIPAssetRegistry
+/// @notice Manages the registration and tracking of Group IPA, including the group members and reward pools.
+abstract contract GroupIPAssetRegistry is IGroupIPAssetRegistry, ProtocolPausableUpgradeable {
+    using IPAccountStorageOps for IIPAccount;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+
+    IGroupingModule public immutable GROUPING_MODULE;
+
+    /// @dev Storage structure for the GroupIPAssetRegistry
+    /// @custom:storage-location erc7201:story-protocol.GroupIPAssetRegistry
+    struct GroupIPAssetRegistryStorage {
+        mapping(address groupIpId => EnumerableSet.AddressSet memberIpIds) groups;
+        mapping(address ipId => address rewardPool) rewardPools;
+        // whitelisted group reward pools
+        mapping(address rewardPool => bool isRegistered) whitelistedGroupRewardPools;
```

Comment:
mapping(address rewardPool => bool isRegistered) whitelistedGroupRewardPools;

can be

mapping(address rewardPool => bool isWhitelisted) whitelistedGroupRewardPools;

Reply:
Renamed.